sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -687,13 +687,48 @@ def get_cmis_media_lanes_mask(self, api, appl, lport, subport):
 
         return media_lanes_mask
 
-    def is_appl_reconfigure_required(self, api, app_new):
+    def set_dp_decommision_pending(self, lport, decomm_pending):
         """
-	   Reset app code if non default app code needs to configured 
+        Set decommission pending flag for the datapath of the given logical port
+        and force CMIS reinit if decomm_pending is False.
+
+        Args:
+            lport:
+                String, logical port name
+            decomm_pending:
+                Boolean, True, if decommission pending is set, False otherwise
+                         False, it means the datapath is ready to be reinitialized.
+        """
+        physical_port = self.port_dict[lport]['index']
+        for logical_port, pdata in self.port_dict.items():
+            if pdata['index'] == physical_port:
+                pdata['is_decomm_pending'] = decomm_pending
+                if not decomm_pending:
+                    self.force_cmis_reinit(logical_port, 0)
+
+    def is_decommission_required(self, lport, api):
```

Comment:
@AnoopKamath please do only what the function name indicate - i.e check for decommsion requited. create more function as needed.